modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2TokenRevokeWithInvalidClientCredentialsTestCase.java

```diff
@@ -0,0 +1,117 @@
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.json.simple.JSONObject;
+import org.json.simple.JSONValue;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.annotations.ExecutionEnvironment;
+import org.wso2.carbon.automation.engine.annotations.SetEnvironment;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+/**
+ * This test case is for IDENTITY-5128, to check that when the oauth2 revoke endpoint is called with invalid client
+ * credentials it should have proper error handling than having a server error.
+ */
+public class OAuth2TokenRevokeWithInvalidClientCredentialsTestCase extends OAuth2ServiceAbstractIntegrationTest {
+
+    private static final String REVOKE_TOKEN_API_ENDPOINT = "https://localhost:9853/oauth2/revoke";
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+        super.init();
+        setSystemproperties();
+        super.init(TestUserMode.SUPER_TENANT_USER);
+    }
+
+    /**
+     * This tests written for test for token revocation after cache timed out CARBON-15028
+     * This test needed two APIM nodes with clustering enabled
+     * During the test one node is use to generate the token and other node use to revoke the token
+     * After cache timeout new token should issued after it revoked
+     *
+     * @throws Exception
+     */
+    @SetEnvironment(executionEnvironments = { ExecutionEnvironment.ALL })
+    @Test(description = "Revoke token with invalid sending of client credentials.")
+    public void testRevokeTokenAfterCacheTimedOut() throws Exception {
+        String clientKey = "dummyConsumerKey";
+        String clientSecret = "dummyConsumerSecret";
+        String accessToken = "dummyAccessToken";
+
+        //Revoke access token
+        ArrayList<NameValuePair> postParameters;
+        HttpClient client = new DefaultHttpClient();
+        HttpPost httpRevoke = new HttpPost(REVOKE_TOKEN_API_ENDPOINT);
+        //Generate revoke token post request
+        httpRevoke.setHeader("Authorization", "Basic " + getInvalidBase64EncodedString(clientKey, clientSecret));
+        httpRevoke.setHeader("Content-Type", "application/x-www-form-urlencoded");
+        postParameters = new ArrayList<NameValuePair>();
+        postParameters.add(new BasicNameValuePair("token", accessToken));
+        httpRevoke.setEntity(new UrlEncodedFormEntity(postParameters));
+        HttpResponse response = client.execute(httpRevoke);
```

Comment:
Can we improve the test case to validate the behavior for different combinations of incorrect client id, client secret and access token?